test/users-query.ts

```diff
@@ -0,0 +1,470 @@
+        totalResults
+        hasUsersBefore
+        hasUsersAfter
+      }
+    }
+  `
+
+  afterEach(async function () {
+    await prisma.user.deleteMany({})
+  })
+
+  it('Should fail due to lack of authentication', async function () {
+    await prisma.user.createMany(reversedList)
+
+    const response = await axios.post('http://localhost:4000', {
+      query: print(usersQuery),
+      variables: {
+        data: {
+          usersPerPage: 10,
+          skippedUsers: 0,
+        },
+      },
+    })
+
+    expect(response.data.data).to.be.null
+    expect(response.data.errors[0].message).to.be.eq('Usuário não autenticado')
+  })
+
+  it('Should return all users on alphabetical order', async function () {
+    await prisma.user.createMany(reversedList)
+
+    const userListWithoutPasswords = userList.data.map((user) => {
+      const { password, ...userWithoutPassword } = user
+      return userWithoutPassword
+    })
+
+    const expectedResponse = {
+      userList: userListWithoutPasswords,
+      totalResults: 10,
+      hasUsersBefore: false,
+      hasUsersAfter: false,
+    }
+
+    const payload = {
+      id: 10000,
+      email: 'payload_email@gmail.com',
+    }
+
+    const token = jwt.sign(payload, process.env.SIGNING_KEY)
+
+    const response = await axios.post(
+      'http://localhost:4000',
+      {
+        query: print(usersQuery),
+        variables: {
+          data: {
+            usersPerPage: 10,
+            skippedUsers: 0,
+          },
+        },
+      },
+      {
+        headers: {
+          Authorization: token,
+        },
+      }
+    )
+
+    expect(response.data.data.users)
+      .excludingEvery('id')
+      .to.be.deep.eq(expectedResponse)
+  })
+
+  it('Should return the first page (users 1-3 from 10 users)', async function () {
+    await prisma.user.createMany(reversedList)
+
+    const userListWithoutPasswords = userList.data.map((user) => {
+      const { password, ...userWithoutPassword } = user
+      return userWithoutPassword
+    })
+
+    const expectedResponse = {
+      userList: [
+        userListWithoutPasswords[0],
+        userListWithoutPasswords[1],
+        userListWithoutPasswords[2],
+      ],
+      totalResults: 10,
+      hasUsersBefore: false,
+      hasUsersAfter: true,
+    }
+
+    const payload = {
+      id: 10000,
+      email: 'payload_email@gmail.com',
+    }
+
+    const token = jwt.sign(payload, process.env.SIGNING_KEY)
+
+    const response = await axios.post(
+      'http://localhost:4000',
+      {
+        query: print(usersQuery),
+        variables: {
+          data: {
+            usersPerPage: 3,
+            skippedUsers: 0,
+          },
+        },
+      },
+      {
+        headers: {
+          Authorization: token,
+        },
+      }
+    )
+
+    expect(response.data.data.users)
+      .excludingEvery('id')
+      .to.be.deep.eq(expectedResponse)
+  })
+
+  it('Should return the second page (users 4-6 from 10 users)', async function () {
```

Comment:
Não sou muito fã de espcificar no enunciado os números certinhos, como 10 users, página de 4 a 6, porque se vc troca esses requisitos nos testes tem que atualizar todos os enunciados. Geralmente eu gosto de de deixar uma descrição que deixa claro do que se trata, mas que não especifique as coisas numericamente. Algo asism:

- should return first page
- should return intermediary page
- should return last page
- should return empty on when requesting to skip more users than the existing total

Mas pode deixar assim, só uma reflexão com relevância não muito alta mesmo.